chat-server/server.js

```diff
@@ -21,10 +20,88 @@ const welcomeMessage = {
 //We will start with one message in the array.
 const messages = [welcomeMessage];
 
+app.put("/messages/:id", (req, res) => {
+  const messageId = parseInt(req.params.id);
+  const { from, text } = req.body;
+
+  const updateMessage = messages.find((msg) => msg.id === messageId);
+  if (!updateMessage) {
+    return res.status(404).json({ error: "Message not found" });
+  }
+  if (from) {
+    updateMessage.from = from;
+  }
+  if (text) {
+    updateMessage.text = text;
+  }
+  res.json(updateMessage);
+});
+
+app.post("/messages", (req, res) => {
+  const { from, text } = req.body;
+  if (!from || !text) {
+    return res
+      .status(400)
+      .json({ error: "Both 'from' and 'text' fields are required." });
+  }
+
+  const newMessage = {
+    id: messages.length,
```

Comment:
This may end up duplicating ID from a previous message (e.g. if you added a message, deleted it, and then added a new one). Can you think why it may be problematic? How could you avoid this issue if you wanted to?